streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -465,6 +465,13 @@ boolean tryToCompleteRestoration(final long now, final java.util.function.Consum
             } catch (final TimeoutException timeoutException) {
                 task.maybeInitTaskTimeoutOrThrow(now, timeoutException);
                 allRunning = false;
+            } catch (final TaskCorruptedException te) {
```

Comment:
Why do we need this?

The idea of the ticket was to change existing code that catches `TaskCorruptionException`: after we wipe out the local state store, we trigger a rebalance if standbys are enabled.